builder/proxmox/common/step_remove_cloud_init_drive.go

```diff
@@ -0,0 +1,92 @@
+package proxmox
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/Telmate/proxmox-api-go/proxmox"
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+// stepRemoveCloudInitDrive removes the cloud-init cdrom and deletes
+// VM config parameters related to cloud-init.
+type stepRemoveCloudInitDrive struct{}
+
+type CloudInitDriveRemover interface {
+	GetVmConfig(*proxmox.VmRef) (map[string]interface{}, error)
+	SetVmConfig(*proxmox.VmRef, map[string]interface{}) (interface{}, error)
+}
+
+var _ CloudInitDriveRemover = &proxmox.Client{}
+
+func (s *stepRemoveCloudInitDrive) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packersdk.Ui)
+	client := state.Get("proxmoxClient").(CloudInitDriveRemover)
+	vmRef := state.Get("vmRef").(*proxmox.VmRef)
+
+	changes := make(map[string]interface{})
+	delete := []string{}
+
+	vmParams, err := client.GetVmConfig(vmRef)
+	if err != nil {
+		err := fmt.Errorf("error fetching template config: %s", err)
+		state.Put("error", err)
+		ui.Error(err.Error())
+		return multistep.ActionHalt
+	}
+	// The cloud-init drive is usually attached to ide0, but check the other controllers as well
+	diskControllers := []string{"ide0", "ide1", "ide2", "ide3"}
+	// Proxmox supports up to 6 SATA controllers (0 - 5)
+	for i := 0; i < 6; i++ {
+		sataController := "sata" + strconv.Itoa(i)
```

Comment:
Same comment here regarding the Sprintf
```suggestion
		sataController := fmt.Sprintf("sata%d", i)
```

Reply:
Done - also for the `ipconfig` case. Thanks again for your review!